test/PR_test/integration_test/summary/test_system.py

```diff
@@ -278,3 +285,87 @@ def instantiate_system():
+            system.network = fe.Network(ops=[
+                TestTensorOp(inputs="x_out", outputs="x_out", mode="train", var1=var1),
+                ModelOp(model=model, inputs="x_out", outputs="y_pred")
+            ])
+            system.traces.append(TestTrace(var1=var1))
+
+            return system
+
+        system = instantiate_system()
+
+        # make some change
+        var1_new_val = 2
+        system.traces[0].var1.assign(var1_new_val)
+
+        # save the state
+        save_path = tempfile.mkdtemp()
+        system.save_state(save_path)
+
+        # reinstantiate system and load the state
+        system = instantiate_system()
+        var1 = system.traces[0].var1
+        system.load_state(save_path)
+
+        with self.subTest("Check both trace and tensorop variables are reloaded"):
+            self.assertEqual(var1_new_val, system.traces[0].var1.numpy())
+            self.assertEqual(var1_new_val, system.network.ops[0].var1.numpy())
+
+        with self.subTest("Check trace and tensorop variables are still shared"):
+            var1_new_val = 3
+            system.traces[0].var1.assign(var1_new_val)
+            self.assertEqual(system.traces[0].var1.numpy(), system.network.ops[0].var1.numpy())
+
+        with self.subTest("Check that variable is still linked to outside code"):
+            var1_new_val = 4
+            var1.assign(var1_new_val)
+            self.assertEqual(system.traces[0].var1.numpy(), system.network.ops[0].var1.numpy())
+
+    def test_shared_torch_variable_among_top_trace(self):
+        def instantiate_system():
+            system = sample_system_object_torch()
+            model = fe.build(model_fn=fe.architecture.pytorch.LeNet, optimizer_fn='adam', model_name='torch')
+            var1 = torch.tensor(1.0)
+            system.network = fe.Network(ops=[
+                TestTensorOp(inputs="x_out", outputs="x_out", mode="train", var1=var1),
+                ModelOp(model=model, inputs="x_out", outputs="y_pred")
+            ])
+            system.traces.append(TestTrace(var1=var1))
+
+            return system
+
+        system = instantiate_system()
+
+        # make some change
+        var1_new_val = 2.0
+        system.traces[0].var1.copy_(torch.tensor(var1_new_val))
+
+        # save the state
+        save_path = tempfile.mkdtemp()
+        system.save_state(save_path)
+
+        # reinstantiate system and load the state
+        system = instantiate_system()
+        var1 = system.traces[0].var1
+        system.load_state(save_path)
+
+        with self.subTest("Check both trace and tensorop variables are reloaded"):
+            self.assertEqual(var1_new_val, system.traces[0].var1.numpy())
+            self.assertEqual(var1_new_val, system.network.ops[0].var1.numpy())
+
+        with self.subTest("Check trace and tensorop variables are still shared"):
+            var1_new_val = 3
+            system.traces[0].var1.copy_(torch.tensor(var1_new_val))
```

Comment:
this won't be needed if you modify lines 347 & 348 to just use the single variable